### ISSUE TITLE
Balance Druid key rendering

### ISSUE THREAD

**Issue by Brdrker** [Sidoine/Ovale#53]
I can't be certain I didn't cause this problem by moving keys around to move more frequently pressed keys closer together. Currently, though, lunar strike, when it shows up (key 4 for me), it does not display the 4 but leaves it blank. Solar Wrath (key 3 for me) shows up labeled as key 4.

On many keys, lower right hand corner, there is a "?" that, for the life of me, I can't figure out. I'm not saying this is a bug but I have no idea why the ? shows up. Note that my Balance druid is only 96.

**Comment by ShmooDude** [comment]
The `?` is your ranged indicator.  It'll turn red when you're out of range for that ability.  You can change it to whatever character you want though in `/ovale config` in the icon section.

Do you have those spells anywhere else on your action bar?  I believe Ovale takes the keybinding for the first version of the ability it finds.  But since the moonkin action bar is pretty much the last on the list, if those spells are any other place, they'll be picked up first.

For example, if I only have Moonfire on my main action bar (non-form at 1) and the main action bar (moonkin form at 2), then it'll work properly.  When you're not in no form it'll put show 1, and when you are in moonkin form it'll show 2.  However, if I put moonfire on another action bar (say 2 which is bottom right in the default interface), then if I'm not in form, it'll still show 1 (because action bar 1 comes before action bar 2) but if I'm in moonkin form, (bar 2 comes before bar 10 which is the moonkin bar) and it shows the keybind for the moonfire on action bar 2, even if that's no keybind at all.

Ovale should probably just keep searching until it finds a keybound version of the spell and prioritize "bonusbars" which are 7-10 over bars 2-6 which is everything but the main action bar (since when in the appropriate form, those are the "main action bar").  Ovale already seems to properly switch to the right keybind if it's bound only on action bar 1 and the bonus bar (at least during my testing).

**Comment by Brdrker** [comment]
The only thing I can figure is that normal character layout on "changing" bar was slightly different from Moonkin "changing" bar. I have two druids, now with the same layout. I just changed the Moonkin and normal to be exactly the same.

Shape shift bar:

Moonfire - Sunfire - Solar Wrath - Lunar Strike - Starsurge - Stellar Flare - Remove Corruption - Renewal - Healing Touch - Typhoon - n/a - n/a

Secondary bar (does not change):

Entangle Roots - Bark Skin - Celestial Alignment - Solar Beam - n/a - Innervate - n/a - n/a - Dash - Prowl - n/a - n/a

Lunar strike is still not picking up that it is key 4 (but muscle memory knows where it is). It's almost like it's immune to your over-write.

Thanks for the explanation on the "?" and, yes, I knew that... I just never registered that this was what it was. But, I did react based on it. Go figure. What I get for being an old fart.

I'll do more testing. As of this morning, the one druid where keys were consistent between normal and moonkin did not seem to have problems (other than the key # display on Lunar Strike. The other druid definitely had some stuff moved around between normal and moonkin. Thinking back on what I saw, I'm now thinking that might be it (and I cannot remember if I was in Moonkin form or normal when I was seeing the oddity. Will advise (and, again, THANKS for doing balance). I had given up on ever doing it again and was thinking I was going to have to go Cat if I wanted to continue to play it. Druids were my first characters :)
